src/BowlingApp.js

```diff
@@ -0,0 +1,110 @@
+$(document).ready(function() {
+  var scorecard = new ScoreCard();
+
+  $("#play_again").click(function() {
+    $("#game_over").hide();
+    location.reload();
+  });
+
+  $("button").click(function(event) {
+    if (scorecard.frame === 11) {
+      gameOver();
+    }
+
+    var score = scorecard.scoreForRoll($(this).data("pins"));
+    var frame = scorecard.frame;
+    $("td[id='" + frame + "']").append('<td>' + score +'</td>');
+    hideButtons(scorecard.pinsLeft(score));
+
+    //check for spare on previous frame
+    if (frame > 1 && scorecard.isPreviousFrameSpare()) {
```

Comment:
Have a look at the "Tell, Don't Ask" principle of OOP for more on why this is undesirable.